Automattic/wp-calypso
# Emails: Add billing interval toggle to In-Depth Comparison page

This pull request adds a selector on the `In-Depth Comparison` page to change the billing interval. It also refines the copy of the header. Finally, this pull request adds `Select` buttons to redirect users to the `Email Comparison` page with the right billing interval selected, and the right email provider expanded:

![image](https://user-images.githubusercontent.com/594356/148405006-08f7e11b-203b-4f43-8e24-8b94b330ba25.png)

Note this pull request also:

* Moves the `BillingIntervalToggle` to its own folder
* Fixes a wrong component name

The following are out-of-scope:

* Using an `Enum` for the email provider slugs
* Handling Google Workspace properly when monthly billing is selected

Regarding the latter I think one option could be to hide the `Select` button and display `unavailable` for now.

#### Testing instructions

Note you will have to append `?flags=emails/in-depth-comparison` to the url of the page (and reload it) if you're testing those changes on a live branch as this feature has not been enabled on this environment yet:

1. Run `git checkout update/in-depth-comparison-page-with-billing-interval` and start your server, or access a [live branch](https://github.com/Automattic/wp-calypso/pull/59803#issuecomment-1006536130)
2. Log into a WordPress.com account with a domain but no email
3. Open the [`Emails` page](http://calypso.localhost:3000/email)
4. Click the `Add Email` button of that domain to access the `Email Comparison` page
5. Change the billing interval
6. Click the `See how they compare` link to access the `In-Depth Comparison` page
7. Assert that the page shows the billing interval you have selected
8. Change the billing interval
9. Click any `Select` button
10. Assert that you are presented with the `Email Comparison` page
11. Assert that the page shows the billing interval you have selected
12. Assert that the form of the email provider selected is expanded (if it exists)

You should also make sure that reloading the page, or going back to a previous page doesn't result in displaying inconsistent information. Here is an example using part of the test instructions from above:

5. Change the billing interval to monthly
6. Click the `See how they compare` link to access the `In-Depth Comparison` page
7. Assert that the page shows the monthly billing interval
8. Click your browser's back button
9. Assert that the billing interval is still monthly
